### ISSUE TITLE
Recovery Seed Phrase can lead to unforeseen issues

### ISSUE THREAD

**Issue by ilblackdragon** [near/near-wallet#1329]
Currently wallet adds new key every time recovery of seed phrase happens.
Given user doesn't understand this, but it also grows number of active keys user have - this can lead to exhausting all $NEAR tokens on storage.

Some crypto users usually would recover every time they want to use the wallet. E.g. use it to "log in".

There should be a way to either use the seed phrase key itself directly (how all the other wallet do it) vs adding new one.
Or at least some explanation of what is happening during this step (not that anyone will read it)

Or at least Log out button to remove the new key, when user doesn't need it (otherwise they won't know they can remove it)

**Comment by vgrichina** [comment]
> Or at least Log out button to remove the new key, when user doesn't need it (otherwise they won't know they can remove it)

yeah, this is probably what we need as people are constantly asking for "Log out" and if implemented this way – it's going to be useful

other than that I think we generally should add some easy way to clean unused keys (requiring user to prove they have recovery key). Can prompt user as part of https://github.com/near/near-wallet/issues/1182

**Comment by kcole16** [comment]
Agree re: removing the keys on logout. 

One UX consideration is that users with 2FA and Ledger enabled will need to explicitly authorize the deleteKey TX, which requires multiple confirmations for users with multiple accounts. 

I believe this is a small % of users currently (is there a straightforward way to pull this data?), though we may want to consider single account logout for the first version of #889.
